src/vos/tests/evt_ctl.c

```diff
@@ -1128,6 +1139,14 @@ test_evt_iter_delete(void **state)
 
 			rc = evt_insert(toh, &entry);
 			assert_int_equal(rc, 0);
+			if (check_um_instance == 0) {
```

Comment:
I wonder if it wouldn't be better to move this logic to utest library ?

Something like this

utest_sync_mem_status(utx); // gets current value and saves it in utx
rc = evt_insert....
rc = utest_check_mem_increase(utx); // gets new value and ensures it's bigger (or at least hasn't decreased)

utest_sync_mem_status(utx);
rc = evt_delete(....)
rc = utest_check_mem_decrease(utx)

the utest_free check could be entirely internal since it's a utest supplied function.

This would also allow us to abstract away the pmem part.   Initially, for VMEM these functions could be noops but if/when we add support for stats there, we could make the change without changing the tests.

Furthermore, this would make doing this for other tests easier.

the utest_check_mem functions could optionally take a a size argument (default 0) to check that things increased/decreased by at least some number of bytes.

Reply:
Nice suggestion. Sure. Will work on it...

Reply:
Done.